planex/makesrpm.py

```diff
@@ -0,0 +1,157 @@
+
+"""
+planex-make-srpm: Wrapper around rpmbuild
+"""
+
+import sys
+import subprocess
+import os
+from shutil import copyfile, rmtree
+import fileinput
+import tarfile
+import tempfile
+
+import argparse
+import argcomplete
+from planex.util import add_common_parser_options
+
+
+def parse_args_or_exit(argv):
+    """
+    Parse command line options
+    """
+    parser = argparse.ArgumentParser(description='Planex SRPM building')
+    add_common_parser_options(parser)
+    parser.add_argument(
+        "--topdir", metavar="DIR", default=None,
+        help='Set rpmbuild toplevel directory')
+    parser.add_argument(
+        "--dist", metavar="DIST", default="",
+        help="distribution tag (used in RPM filenames)")
+    argcomplete.autocomplete(parser)
+    return parser.parse_known_args(argv)
+
+
+def parse_patchseries(series_file):
+    """
+    Parse series file and return the list of patches
+    """
+    patches = []
+    with open(series_file) as series:
+        for line in series:
+            line = line.partition('#')[0].strip()
+            if line:
+                patches.append(line)
+
+    return patches
+
+
+def setup_tmp_area():
+    """
+    Create temporary working area
+    """
+    tmp_dirpath = tempfile.mkdtemp()
+    tmp_specs = os.path.join(tmp_dirpath, 'SPECS')
+    tmp_build = os.path.join(tmp_dirpath, '_build')
+    tmp_sources = os.path.join(tmp_build, 'SOURCES')
+
+    os.makedirs(tmp_specs)
+    os.makedirs(tmp_build)
+    os.makedirs(tmp_sources)
+
+    return (tmp_dirpath, tmp_sources)
+
+
+def extract_patches(tmp_specfile, patchqueue_filters, patchqueue_path,
+                    tmp_sources, target):
+    """
+    Extract patches and inject them into the specfile
+    """
+    for line in fileinput.input(tmp_specfile, inplace=True):
+        if any([ext in line for ext in patchqueue_filters]):
+            tar = tarfile.open(patchqueue_path)
+            for mem in tar.getmembers():
+                # Modify mem.name in place to change the name of the
+                # extracted file (and drop any leading paths)
+                mem.name = '%s-' % target + os.path.basename(mem.name)
+                tar.extract(mem, tmp_sources)
+            patches = parse_patchseries(os.path.join(tmp_sources,
+                                                     '%s-series' % target))
+            print "# Patches for %s" % target
+            for patch_num, patch in enumerate(patches):
+                if patch != "":
+                    print "Patch%s: %%{name}-%s" % (patch_num, patch)
+        else:
+            print line,
+
+
+def get_command_line(intercepted_args, tmp_sources, tmp_specfile):
+    """
+    Return rpmbuild command line and arguments
+    """
+    cmd = ['rpmbuild']
+    if intercepted_args.quiet:
+        cmd.append('--quiet')
+    if intercepted_args.topdir is not None:
+        cmd.append('--define')
+        cmd.append('_topdir %s' % intercepted_args.topdir)
+    if intercepted_args.dist != "":
```

Comment:
If you change the default above, this should become `if intercepted_args.dist is not None:`